cloud-functions/comparer-function/main.py

```diff
@@ -137,15 +139,33 @@ def publish_message(project_id, topic_name, message):
     future = publisher.publish(topic_name, b'hello from compare func')
     future.result()
 
+# define the function to compute Mean Squared Error between two images
+def meanSquaredError(img1, img2):
+   height, width = img1.shape
+   diff = cv2.subtract(img1, img2)
+   err = np.sum(diff**2)
+   mse = err/(float(height*width))
+   return mse, diff
+
+def compare(image_a, image_b, imageMask):
+    "compare two images and return the percentage of difference"
+
+    #Read images
+    image_a_data = cv2.imread(image_a)
+    image_b_data = cv2.imread(image_b)
+
+    #Converting images to GrayScale
+    image_a_data = cv2.cvtColor(image_a_data, cv2.COLOR_BGR2GRAY)
+    image_b_data = cv2.cvtColor(image_b_data, cv2.COLOR_BGR2GRAY)
 
-def compare(image_a, image_b):
-    """compare two images and return the percentage of difference"""
+    #Check if image Mask is not none and apply
+    if imageMask is not None:
+        mask = cv2.imread('mask.png',0)
```

Comment:
- I would pass in `imageMask` as the the file path value.

Reply:
Was passed just missed update of the read